dev.c

```diff
@@ -554,8 +558,41 @@ static int fuse_notify_add_ext(struct fuse_conn *conn, unsigned int size,
 		printk(KERN_ERR "%s: can't copy arg\n", __func__);
 		return -EFAULT;
 	}
+
+	trace_fuse_notify_add_ext(add.dev_id, add.size, add.queue_depth, add.discard_size, add.open_mode, add.enable_fp, add.paths.count);
+	return pxd_add(conn, &add, false);  // v1 structure
+}
+
+static int fuse_notify_add_ext_v2(struct fuse_conn *conn, unsigned int size,
+		struct iov_iter *iter)
+{
+	struct pxd_add_ext_out_v2 add;
+	// Calculate size without padding: offset of last field + size of last field
+	size_t expected_size = offsetof(struct pxd_add_ext_out_v2, reserved) + sizeof(uint64_t);
+	size_t len;
+
+	if (size < expected_size) {
+		printk(KERN_ERR "%s: invalid size %u (expected %zu)\n", __func__, size, expected_size);
+		return -EINVAL;
+	}
+
+	len = expected_size;
+	memset(&add, 0, sizeof(add));  // Zero-initialize to handle any padding
+
+	if (copy_from_iter(&add, len, iter) != len) {
+		printk(KERN_ERR "%s: can't copy arg (expected %zu bytes)\n", __func__, len);
+		return -EFAULT;
+	}
+
+	// Validate reserved field is zero (for future compatibility)
+	if (add.reserved != 0) {
+		printk(KERN_WARNING "%s: reserved field is non-zero (%llu), ignoring\n",
+		       __func__, add.reserved);
+	}
+
+	// Cast to pxd_add_ext_out for pxd_add() - the extra fields are at the end
 	trace_fuse_notify_add_ext(add.dev_id, add.size, add.queue_depth, add.discard_size, add.open_mode, add.enable_fp, add.paths.count);
-	return pxd_add(conn, &add);
+	return pxd_add(conn, (struct pxd_add_ext_out *)&add, true);  // v2 structure
```

Comment:
This is wrong... pxd_add() should use pxd_add_v2_out structure as input and all existing interface should convert to this.